onnxruntime/core/optimizer/embed_layer_norm_fusion.cc

```diff
@@ -165,17 +166,72 @@ Status EmbedLayerNormFusion::ApplyImpl(Graph& graph, bool& modified, int graph_l
       continue;
     }
 
-    // Match Shape --> Expand path if needed.
-    std::vector<graph_utils::EdgeEndToMatch> position_embedding_path_symbolic{
-        {0, 1, "Expand", {8}, kOnnxDomain},
-        {0, 1, "Shape", {1}, kOnnxDomain}};
+    // Check the second input of position gather. If it's not initializer, check for two paths.
     Node* p_expand_node = nullptr;
     Node* p_shape_node = nullptr;
-    if (graph_utils::FindPath(position_gather_node, true, position_embedding_path_symbolic, edges, logger)) {
-      if (edges[0]->GetNode().GetOutputEdgesCount() == 1 && edges[1]->GetNode().GetOutputEdgesCount() == 1) {
-        p_expand_node = graph.GetNode(edges[0]->GetNode().Index());
-        p_shape_node = graph.GetNode(edges[1]->GetNode().Index());
+    std::vector<const Node::EdgeEnd*> pg_edges;
+    bool isValidEmbedSubNode = true;
+    if (graph_utils::IsConstantInitializer(graph, position_gather_node.MutableInputDefs()[1]->Name())) {
+      // Check if the second input of position gather is a tensor with values evenly spaced by 1 starting from 0. 
+      std::vector<int64_t> data;
+      if (!optimizer_utils::AppendTensorFromInitializer(graph, *(position_gather_node.MutableInputDefs()[1]), data) ||
+          data.size() < 1 ||
```

Comment:
data.size() shall be same as dimension 1 of 3 graph inputs.

Reply:
Should data.size() be batch_size * sequence length (if batch size is not 1)?

Reply:
Right, I think the shape is [batch,seq_length]. So the logic is a little complicated than current implementation.